src/v/archival/segment_reupload.cc

```diff
@@ -346,22 +358,22 @@ segment_collector::make_upload_candidate(
     auto locks_resolved = co_await ss::when_all_succeed(
       locks.begin(), locks.end());
 
-    auto starting_offset = head_seek_result.offset;
+    auto starting_offset = head_seek.offset;
     if (starting_offset != _begin_inclusive) {
         vlog(
           archival_log.info,
           "adjusting begin offset of upload candidate from {} to {}",
-          head_seek_result.offset,
+          head_seek.offset,
```

Comment:
nit: I would use starting_offset here, or use head_seek.offset in the condition.
same for final_offset later this file

Reply:
good point, fixed